setup.py

```diff
@@ -21,26 +29,31 @@ def rtd_dependent_deps():
     packages=find_packages(exclude=['tests', 'tests.*']),
     python_requires='>=3.6',
     install_requires=[
-        'capstone>=3.0.5',
-        'pyelftools',
-        'unicorn',
-        'ply',
-        'pysha3',
-        # In 0.1.1, pyevmasm changed its gas cost calculations, so Manticore will need to update its
-        # unit tests to match before we can upgrade pyevmasm
-        'pyevmasm==0.1.0',
         'pyyaml',
     ] + rtd_dependent_deps(),
     dependency_links=[
         'https://github.com/aquynh/capstone/archive/next.zip#egg=capstone-4&subdirectory=bindings/python',
     ],
     extras_require={
+        'native': [
+            'capstone>=3.0.5',
+            'pyelftools',
+            'unicorn',
+            'ply',
+        ],
+        'evm': [
+            'pysha3',
+            # In 0.1.1, pyevmasm changed its gas cost calculations, so Manticore will need to update its
+            # unit tests to match before we can upgrade pyevmasm
+            'pyevmasm==0.1.0',
```

Comment:
evm will need ply for the ABI parser I think

Reply:
So it's a common dependency, thanks. I haven't run tests for those things yet.

I will have to move some files/classes/exceptions here and there to make things work properly.

I am thinking of creating `manticore/native` directory and move there e.g. `manticore/core/cpu`, `manticore/binary`. Not sure about:
* `manticore/core/memory.py` -  but it doesn't use any native dependencies, so I think I will keep it for now
* `manticore/core/plugin.py` - this will have to be split as it has some native plugins inside
* `manticore/core/state.py` - it uses `from .cpu.abstractcpu import ConcretizeRegister` which has capstone dependency, so I will have to split it somehow. It uses it in `class State(Eventful)` in `def execute` :S.

Reply:
I'd say:
* manticore/core/memory.py -> native
*manticore/core/plugin.py -> split +1
* manticore/core/state.py -> refactor. ConcretizeRegister is not fo abstract State. Maybe we need AbstractState, StateLinux and StateEVM ?